evm_loader/program/src/account/state.rs

```diff
@@ -1,13 +1,15 @@
 use std::cell::{Ref, RefMut};
-use std::collections::BTreeMap;
-use std::mem::size_of;
+use std::mem::{size_of, ManuallyDrop};
+use std::ptr::addr_of_mut;
 
-use crate::account_storage::AccountStorage;
+use crate::account_storage::{AccountStorage, ProgramAccountStorage};
```

Comment:
I suggest using conditional compilation with `#[cfg(target_os = "solana")]` for code used only in NeonEVM Solana program, until at least the feature of this PR is implemented. Then the code can be decoupled more to reduce the usages of `#[cfg(target_os = "solana")]`.

```Rust
use crate::account_storage::AccountStorage;
#[cfg(target_os = "solana")]
use crate::account_storage::ProgramAccountStorage;
```

Reply:
Thanks for the feedback! As discussed on the meeting, I went with a different path: split ExecutorState into persistent part (struct with all data owned, now called ExecutorStateData) and into refs (old Executor State, mutably borrows ExecutorState).